test/parallel/xyk-pallet.API.poolLiqidity.mintLiqudity.test.ts

```diff
@@ -136,3 +142,57 @@ test("Check that a user that mints on a non-promoted pool liquidity tokens are f
   expect(differenceLiqTokensFree).bnEqual(defaultCurrencyValue);
   expect(differenceLiqTokensReserved).bnEqual(new BN(0));
 });
+
+test("Given 3 pool: token1-MGX, token2-MGX and token1-token2 WHEN token1-token2 is promoted THEN user can receive rewards from token1-token2 pool", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(token1, testUser1, Assets.DEFAULT_AMOUNT),
+    Assets.mintToken(token2, testUser1, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser1),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.createPool(
+        token1,
+        Assets.DEFAULT_AMOUNT.divn(2),
+        token2,
+        Assets.DEFAULT_AMOUNT.divn(2)
+      )
+    )
+  );
+
+  const liqIdThirdPool = await getLiquidityAssetId(token1, token2);
+  const rewardsThirdPoolBefore = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIdThirdPool
+  );
+  await Sudo.batchAsSudoFinalized(
+    Xyk.updatePoolPromotion(liqIdThirdPool, 20),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.mintLiquidity(
+        token1,
+        token2,
+        defaultCurrencyValue,
+        new BN(Number.MAX_SAFE_INTEGER)
+      )
+    )
+  );
+  await waitForRewards(testUser1, liqIdThirdPool, 21);
+  const mangata = await getMangataInstance(
+    getEnvironmentRequiredVars().chainUri
+  );
+  const testUser1Rewards = await mangata.calculateRewardsAmount(
+    testUser1.keyRingPair.address,
+    liqIdThirdPool.toString()
+  );
+  await claimRewardsAll(testUser1, liqIdThirdPool);
+
+  const rewardsThirdPoolAfter = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIdThirdPool
+  );
+
+  expect(rewardsThirdPoolBefore.activatedAmount).bnEqual(BN_ZERO);
+  expect(rewardsThirdPoolAfter.activatedAmount).bnEqual(defaultCurrencyValue);
+  expect(rewardsThirdPoolBefore.rewardsAlreadyClaimed).bnEqual(BN_ZERO);
+  expect(testUser1Rewards).bnLte(rewardsThirdPoolAfter.rewardsAlreadyClaimed);
```

Comment:
Additionally,
Plz either ensure that the user got tokens from the claimRewardsAll or validate that testUser1Rewards is gt BN_Zero